crates/adapters/src/controller/mod.rs

```diff
@@ -826,6 +828,83 @@ impl OutputEndpoints {
     }
 }
 
+/// Buffer used by the output endpoint thread to accumulate outputs.
+struct OutputBuffer {
+    endpoint_name: String,
+
+    buffer: Option<Box<dyn SerTrace>>,
+
+    /// Step number of the last update in the buffer.
+    ///
+    /// The endpoint will wait for this step to commit before sending the buffer
+    /// out.
+    buffered_step: Step,
+
+    /// Time when the first batch was pushed to the buffer.
+    buffer_since: Instant,
+
+    /// Number of input records that will be fully processed after the buffer is flushed.
```

Comment:
what is "processed"? input to what?

Reply:
This is part of the progress tracking mechanism, which tracks the number of inputs to the pipeline that have been processed to completion. I'll clarify the comment.